### DISCUSSION TITLE
Help with entering the path of field map images

### DISCUSSION THREAD

**Discussion by sunyunseung1** [HALFpipe/HALFpipe#267] [Q&A] [answered]
Dear all,

I am trying to enter the path of field map images, but it does not respond when I specify the path and press enter.
In detail, I am trying to use EPI (blip-up blip-down) image.

In addition, when using blip-up blip-down image, do I have to put field maps having different direction separately or merged one?
For example, fsl topup command requires merged image of different directions.
It would be very helpful if I can get any information on data preparation for using field maps. 

I attach a screenshot showing where I am having trouble. 
(It does not respond further from the last step shown in the screenshot.)
I would very thankful for any advice or comments.
![halfpipe_fieldmap_error](https://user-images.githubusercontent.com/99384107/153362168-739e2edd-6605-4ef2-8b46-59ac0ae3c92e.png)

Thanks
Sun.

**Comment by HippocampusGirl** [comment]
Hi Sun,

You will want to have separate files for each direction and name them so that each can be identified with the phase encoding direction in the file name. For example, you could create the files `REST_LR.nii.gz` and `REST_RL.nii.gz` for each participant. Then you can specify the file name pattern as `.../{subject}/REST/REST_{direction}.nii.gz`.

Best,
Lea

**Comment by sunyunseung1** [comment, marked as answer]
Hi Lea,

Thank you for your information. 
I am trying again with separate files for each direction, however, it still does not respond.
So I checked the err.txt, and I could find error messages related to the step in the file. 
I tried a few things but I am still getting the same error message. 
I attach the err.txt.
I would be very thankful for any advice to figure out what the problem is and how I can solve it. 

Thanks,
Sun
[err.txt](https://github.com/HALFpipe/HALFpipe/files/8122521/err.txt)

**Reply by HippocampusGirl** [reply]
Hi Sun,

Of course, I'm happy to help :-)

Could you let me know which version of HALFpipe you are using?

Best,
Lea

**Reply by sunyunseung1** [reply]
Thank you:)
I am using 'halfpipe_1.1.1'.

Best,
Sun

**Reply by HippocampusGirl** [reply]
Hi Sun,

That is good to know. The issue you're experiencing was fixed in version 1.2.1 (see https://github.com/HALFpipe/HALFpipe/commit/d5acf471d9e3fa4b23db808946afbc3a8bfccdf2), so I would recommend upgrading. If you are just starting out with analyses, it should be quite straight-forward to just download the new container and try again :-)

Best,
Lea

**Reply by sunyunseung1** [reply]
Hi Lea,

I tried with version 1.2.1, and now it's working. Thank you!
I have one more question.
Does halfpipe use phase encoding direction info in the file name (LR or RL), not reading the header, when processing field maps?
You told me each file is identified with phase encoding direction in the file name.
I want to make sure that this means halfpipe gets the direction info from the file name when it processes the field maps.
Thanks a lot.

Best,
Sun

**Reply by HippocampusGirl** [reply]
Hi Sun, 

This depends. For functional images, the phase encoding direction can be read from the image header or a sidecar JSON file. 

If you are doing blip-up blip-down distortion correction, then you need to have the direction in the file name.

Hope this helps :-)

Best,
Lea

**Reply by sunyunseung1** [reply]
Hi Lea,

It helped a lot!
Thanks for your reply:)

Best,
Sun
